core/new-gui/src/app/workspace/component/operator-panel/operator-label/operator-label.component.spec.ts

```diff
@@ -67,6 +70,64 @@ describe('OperatorLabelComponent', () => {
   });
 
+  it('should call the mouseEnter function once the cursor is hovering above a operator label', () => {
+    const spy = spyOn<any>(component, 'mouseEnter');
+    const operatorLabelElement = fixture.debugElement.query(By.css('#' + component.operatorLabelID));
+    operatorLabelElement.triggerEventHandler('mouseenter', component);
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should call the mouseLeave function once the cursor leaves a operator label', () => {
+    const spy = spyOn<any>(component, 'mouseLeave');
+    const operatorLabelElement = fixture.debugElement.query(By.css('#' + component.operatorLabelID));
+    operatorLabelElement.triggerEventHandler('mouseleave', component);
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should emits a command to open an tooltip instance after 500ms delay', marbles((m) => {
+    const expectedStream = m.hot('500ms -a-');
+    const actualStream = component.getopenCommandsStream().map(() => 'a');
+    m.hot('-a-').do(() => component.mouseEnter()).subscribe();
+    m.expect(actualStream).toBeObservable(expectedStream);
+  }));
+
+  it('should display a tooltip instance with the correct content when openCommandObservable$ emits a value', marbles((m) => {
+    const operatorLabelElement = fixture.debugElement.query(By.css('#' + component.operatorLabelID));
+    component.getopenCommandsStream().subscribe(x => {
+      if (operatorLabelElement.parent) {
+        const tooltipInstance = operatorLabelElement.parent.childNodes[1].nativeNode;
+        expect(tooltipInstance.innerText).toBe(mockOperatorData.additionalMetadata.operatorDescription);
+      }
+    });
+    m.hot('-a-').do(() => component.mouseEnter()).subscribe();
+  }));
+
+  it('should not emits a command to open tooltip instance if the cursor has left the operator label', marbles((m) => {
+    const expectedStream_1 = m.hot('-----');
+    const actualStream_1 = component.getopenCommandsStream().map(() => 'a');
+    m.hot('-a-----').do(() => component.mouseEnter()).subscribe();
+    m.hot('---b---').do(() => component.mouseLeave()).subscribe();
+    m.expect(actualStream_1).toBeObservable(expectedStream_1);
+  }));
+
+  it('should hide the tooltip instance if cursor leaves the operator label', marbles((m) => {
+    const operatorLabelElement = fixture.debugElement.query(By.css('#' + component.operatorLabelID));
+    component.getopenCommandsStream().subscribe(() => {
+      if (operatorLabelElement.parent) {
+        const tooltipInstance = operatorLabelElement.parent.childNodes[1].nativeNode;
+        expect(tooltipInstance).not.toBeNull();
+      }
+    });
+    m.hot('-a-').do(() => component.mouseEnter()).subscribe();
+    // at this moment, the tooltip is open
+    // it will be closed in the following lines
+    m.hot('600ms b-').do(() => component.mouseLeave()).subscribe(() => {
+      if (operatorLabelElement.parent) {
+        expect(operatorLabelElement.parent.childNodes.length).toBe(1);
```

Comment:
`operatorLabelElement.parent` should always be true when doing this test. We would want to capture the case when `operatorLabelElement.parent === undefined` and throw an error. This design ensures that we capture the error as soon as possible.

One thing that you can do is:
`const parent = operatorLabelElement.parent`
`if (!parent) { expect(true).toBeFalsy(); return; }`
`expect(operatorLabelElement.parent.childNodes.length).toBe(1);`

By doing this, the test will immediately throw error if parent is undefined.

Reply:
Ok, I will change it now.